perun-core/src/main/java/cz/metacentrum/perun/core/impl/UsersManagerImpl.java

```diff
@@ -1462,7 +1462,7 @@ private String getSQLSelectForUsersPage(UsersPageQuery query) {
 				" FROM users";
 
 		String selectWithMembers =
-			"SELECT " + userMappingSelectQuery +
+			"SELECT distinct " + userMappingSelectQuery +
```

Comment:
Wasn't removing `distinct` our great speed optimization when searching users/members? It was probably in a different place, but maybe we should think about the way we select data.

Reply:
I believe it was in member-group search, I will think more about it and revise it if I find a better way...

Reply:
I didn't find a better way. I think we need each of the user's member, so we can find it by searchString, and we want to group them up at the end of the query...